### DISCUSSION TITLE
force_forward与force_ttl优先级问题

### DISCUSSION THREAD

**Discussion by lr7skquo** [kkkgo/PaoPaoDNS#76] [问答和帮助] [answered]
请问如果希望example.com以及其全部子域名（排除cn.example.com）走ppgw网关，应该怎么设置？
按如下设置后，cn.example.com还是解析为fake-ip.
force_forward_list.txt:
```
domain:example.com
```
force_ttl_rules.txt:
```
cn.example.com@111.111.111.111,222.222.222.222
#111.111.111.111,222.222.222.222是example.com的NS服务器IP
```

使用场景是DDNS服务商网站被墙，但动态域名（子域名）可以正确解析。

**Comment by kkkgo** [comment, marked as answer]
文档说明：匹配优先级`force_forward_list > force_nocn_list > force_cn_list`
force_ttl_rules优先级位于以上列表最后。

首先，如果他是境外IP，如果在开了AUTO_FORWARD的情况下本身就已经被forward，那么只需要额外添加force_nocn_list （因为境外域名不一定能递归成功。），或者在对NS服务静态路由到指定网关后，使用force_ttl_rules指定NS服务器。

然后如果没有使用AUTO_FORWARD的话，可以使用custom_mod功能，该功能可以自定义优先级。

**Reply by lr7skquo** [reply]
感谢提供方案。没有使用AUTO_FORWARD，通过custom_mod搞定了。

**Comment by kkkgo** [comment]
顺便DDNS可以申请一个免费的域名，比如eu.org之类，然后托管在Cloudflare或者Dnspod，本机用DDNS脚本；
推荐一下DDNS项目：https://github.com/kkkgo/UE-DDNS
